pkg/maps/ctmap/ctmap_privileged_test.go

```diff
@@ -836,3 +860,115 @@ func (k *CTMapPrivilegedTestSuite) TestOrphanNatGC(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(len(buf), Equals, 0)
 }
+
+// TestCount checks whether the CT map batch lookup dumps the count of the
+// entire map reliably, while the map is changing in the background.
+func (k *CTMapPrivilegedTestSuite) TestCount(c *C) {
+	t := c.T
+
+	// Set the max size of the map explicitly so we can provide enough buffer
+	// for the LRU map to avoid eviction that makes the assertions within this
+	// test indeterministic and consequently cause flakes.
+	prev := option.Config.CTMapEntriesGlobalTCP
+	defer func() { option.Config.CTMapEntriesGlobalTCP = prev }()
+	option.Config.CTMapEntriesGlobalTCP = 4096 << 7
+	size := 4096 << 2
+
+	m := newMap(MapNameTCP4Global+"_test", mapTypeIPv4TCPGlobal)
+	err := m.OpenOrCreate()
+	assert.NoError(t, err)
+	defer assert.NoError(t, m.Map.Unpin())
+
+	cache := populateFakeDataCTMap4(m, size)
+	initial := len(cache)
+
+	start, done := make(chan struct{}), make(chan struct{})
+	toDelete := 4096 << 2
+	failedToDelete := 0
+	go func() {
+		c.Helper()
+
+		for i := 0; i < toDelete; i++ {
+			if i == 0 { // close the channel once at the first iteration.
+				close(start)
+			}
+			k := mapsexp.Keys(cache)[0]
+			if err := m.Delete(k); err != nil {
+				failedToDelete++
+				continue
+			}
+			delete(cache, k)
+		}
+		close(done)
+	}()
+
+	<-start
+out:
+	for {
```

Comment:
What's the reason for calling Count() in a loop? Is the result ever expected to exceed `initial`?